src/app/core/utils/api-token/api-token.service.ts

```diff
@@ -62,121 +63,140 @@ const DEFAULT_EXPIRY_TIME = 3600000;
 
 @Injectable({ providedIn: 'root' })
 export class ApiTokenService {
-  apiToken$ = new ReplaySubject<string>(1);
+  apiToken$: Subject<string>;
   cookieVanishes$ = new Subject<ApiTokenCookieType>();
 
   private cookieOptions: CookieOptions = {};
 
-  private initialCookie$: Observable<ApiTokenCookie>;
+  private initialCookie$: Subject<ApiTokenCookie>;
 
-  constructor(private cookiesService: CookiesService, private store: Store, appRef: ApplicationRef) {
+  private cookieChangeEvent$: Observable<[ApiTokenCookie, ApiTokenCookie]>;
+
+  constructor(private cookiesService: CookiesService, private store: Store, private appRef: ApplicationRef) {
     // setup initial values
     const initialCookie = this.parseCookie();
-    this.initialCookie$ = of(!SSR ? initialCookie : undefined);
-    this.initialCookie$.pipe(mapToProperty('apiToken')).subscribe(token => {
-      this.apiToken$.next(token);
-    });
+    this.initialCookie$ = new BehaviorSubject<ApiTokenCookie>(!SSR ? initialCookie : undefined);
+    this.apiToken$ = new BehaviorSubject<string>(initialCookie?.apiToken);
 
     if (!SSR) {
-      // save token routine
-      combineLatest([
-        store.pipe(select(getLoggedInUser), startWith(undefined), pairwise()),
-        store.pipe(select(getCurrentBasket)),
-        store.pipe(select(getSelectedOrderId)),
-        this.apiToken$,
-      ])
-        .pipe(skip(1), this.mapToApiTokenCookie(), distinctUntilChanged<ApiTokenCookie>(isEqual))
-        .subscribe(apiToken => {
-          const cookieContent = apiToken?.apiToken ? JSON.stringify(apiToken) : undefined;
-          if (cookieContent) {
-            cookiesService.put('apiToken', cookieContent, {
-              expires: this.cookieOptions?.expires ?? new Date(Date.now() + DEFAULT_EXPIRY_TIME),
-              secure: this.cookieOptions?.secure ?? true,
-              sameSite: 'Strict',
-              path: '/',
-            });
-          } else {
-            cookiesService.remove('apiToken');
-          }
-        });
-
-      // access token vanishes routine
-      appRef.isStable
-        .pipe(
-          whenTruthy(),
-          first(),
-          mergeMap(() =>
-            interval(1000).pipe(
-              map(() => this.parseCookie()),
-              pairwise(),
-              // trigger only if application token exists but external token vanished
-              withLatestFrom(this.apiToken$),
-              filter(([[previous, current], apiToken]) => !!previous && !current && !!apiToken),
-              map(([[previous]]) => previous.type)
-            )
-          )
+      this.cookieChangeEvent$ = this.checkCookieRoutine().pipe(shareReplay(1));
+      this.saveTokenRoutine();
+      this.logoutRoutine();
+      this.tokenVanishRoutine();
+      this.tokenCreatedOnAnotherTabRoutine();
+    }
+  }
+
+  private checkCookieRoutine() {
```

Comment:
remove 'Routine' from all function names, add return types and add a short description on top of the functions - what do they do?